### DISCUSSION TITLE
Persisting with SolidJS and Svelte with new coming version 5?

### DISCUSSION THREAD

**Discussion by frederikhors** [TanStack/query#5463] [Q&A]
I would like to know if with the new coming version 5 it will be possible to persist data in `LocalStorage` or `IndexedDB` both using Svelte and SolidJS.

This is the main block we now have for the transition to @tanstack/query.

Thanks in advance for the answer and this amazing project!

**Comment by TkDodo** [comment]
the only thing that's missing is a svelte / solid adapter based on `query-persist-client-core`. Have a look at the react version, it's just 50 LOC:

https://github.com/TanStack/query/blob/f57c8dc15ec7cc20c674892eacd7596a4df39085/packages/react-query-persist-client/src/PersistQueryClientProvider.tsx

So, what we'd need is:
- a way to trigger the subscription to the external storage via `persistQueryClient`
- a way to respect `isRestoring`
- a call to `onSuccess`
- a way to render two providers

if that feature is important to you, maybe you can consider contributing it? You could also start out just using the `query-persist-client-core` package in your app, the framework adapter is really just a small, optional thing.

@ardeora @lachlancollins FYI

**Reply by frederikhors** [reply]
I really wish I was good enough to write this code. But I'm afraid I'm not up to it. Is it possible that no one has thought of it yet? Is it possible that no one has needed this yet?

**Reply by aadito123** [reply]
@TkDodo I would be happy to work on the Solid implementation for this if it is not being worked on by core members. Should this just be a standard PR with a new `solid-query-persist-client` package? Would appreciate any other advice you might have as well.

**Reply by TkDodo** [reply]
yes, you can work on it. @ardeora FYI.

`solid-query-persist-client` would be a new package, regarding setup and dependencies and stuff, just look at how `react-query-persist-client`.

please use the `beta` branch (v5) as base.
